pkg/services/ngalert/eval/eval.go

```diff
@@ -334,7 +334,7 @@ func buildDatasourceHeaders(ctx context.Context, metadata map[string]string) map
 	headers := make(map[string]string, len(metadata)+3)
 
 	for key, value := range metadata {
-		headers[fmt.Sprintf("X-Rule-%s", key)] = url.QueryEscape(value)
+		headers[fmt.Sprintf("http_X-Rule-%s", key)] = url.QueryEscape(value)
 	}
```

Comment:
this changes how it is sent to other data sources, doesn't it?

Reply:
Yea, it is the standard per se. This part in the plugin-sdk will pull headers with `http_` prefixed and add them: https://github.com/grafana/grafana-plugin-sdk-go/blob/c5aa822be58253fd954e921aec00241833d9ca4c/backend/http_headers.go#L74-L77